### DISCUSSION TITLE
Error when attempting to read or load branch-fitted Stan models

### DISCUSSION THREAD

**Discussion by lucumas** [ropensci/targets#791] [Help] [answered]
**Expected behavior:** Calling `tar_read` on a particular branch of fitted Stan models lets me interact with the object the same way I would with a regular non-branch target (I have verified the code below works as expected if the `pattern` argument is dropped).

**Actual behavior:** Error message when calling `tar_read` or `tar_load` on the fitted object (`mtcars_fit`) seemingly related to it being un-vectorizable.

I should note that I have already seen the `stantargets` examples and am aware that doing some additional processing inside the target produces an object that behaves nicely, I'd just like to understand what's going on in this case and how to retrieve the information, as I've already completed a somewhat lengthy set of simulations in this manner.

Reprex below. Apologies if this isn't the correct place to post this as it doesn't exclusively pertain to `targets`.

``` r
library(targets)
tar_script({
    library(targets)
    library(cmdstanr)
    write_stan_file("data {int<lower=0> N; vector[N] x; vector[N] y;}\n  parameters {real alpha; real beta; real<lower=0> sigma;}\n  model {y ~ normal(alpha + beta * x, sigma);}", 
        dir = ".", basename = "linreg")
    fit_stan <- function(model, data) {
        model$sample(data = data, parallel_chains = 2, refresh = 0, 
            show_messages = FALSE, output_dir = ".")
    }
    list(tar_target(itertar, 1:2), tar_target(linreg_model, cmdstan_model("linreg.stan")), 
        tar_target(mtcars_list, list(N = nrow(mtcars), x = mtcars$mpg, 
            y = mtcars$disp)), tar_target(mtcars_fit, fit_stan(linreg_model, 
            mtcars_list), pattern = map(itertar)))
})

tar_make()
#> * start target itertar
#> * built target itertar
#> * start target linreg_model
#> * built target linreg_model
#> * start target mtcars_list
#> * built target mtcars_list
#> * start branch mtcars_fit_f90e102e
#> * built branch mtcars_fit_f90e102e
#> * start branch mtcars_fit_187f54c1
#> * built branch mtcars_fit_187f54c1
#> * built pattern mtcars_fit
#> * end pipeline

tar_read(mtcars_fit, branches = 1)
#> Error in `stop_vctrs()`:
#> ! `mtcars_fit_f90e102e` must be a vector, not a <CmdStanMCMC/CmdStanFit/R6> object.
```

**Comment by wlandau** [comment, marked as answer]
`tar_target(mtcars_fit, ..., iteration = "list")` should fix the issue. That way, `targets` will not try to put the Stan models into a vector. Related: https://books.ropensci.org/targets/dynamic.html#list-iteration

**Reply by lucumas** [reply]
Makes sense, thank you!
